apps/researcher/src/components/form/edtf/converter.test.ts

```diff
@@ -0,0 +1,83 @@
+import {expect} from '@jest/globals';
+import {fromDateString, toDateString} from './converter';
+
+describe('toDateString', () => {
+  it('returns an empty string when no date components are provided', () => {
+    const result = toDateString();
+    expect(result).toEqual('');
+  });
+
+  it('returns the correct date string when all date components are provided', () => {
+    const yyyy = '2022';
+    const mm = '10';
+    const dd = '15';
+    const result = toDateString(yyyy, mm, dd);
+    expect(result).toEqual('2022-10-15');
+  });
+
+  it('returns the correct date string when only year and month are provided', () => {
+    const yyyy = '2022';
+    const mm = '10';
+    const result = toDateString(yyyy, mm);
+    expect(result).toEqual('2022-10');
+  });
+
+  it('returns the year when only year and day are provided', () => {
+    const yyyy = '2022';
+    const dd = '15';
+    const result = toDateString(yyyy, undefined, dd);
+    expect(result).toEqual('2022');
+  });
+
+  it('returns the correct date string when only year is provided', () => {
+    const yyyy = '2022';
+    const result = toDateString(yyyy);
+    expect(result).toEqual('2022');
+  });
+
+  it('returns an negative year if year is negative', () => {
+    const yyyy = '-500';
+    const result = toDateString(yyyy);
+    expect(result).toEqual('-0500');
+  });
+
+  it('returns the correct date string when year is negative and other components are provided', () => {
```

Comment:
Perhaps be explicit about 'other components'? (what are these, and what are 'components'?)